flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/delegation/BatchPlanner.scala

```diff
@@ -134,6 +134,11 @@ class BatchPlanner(
     new BatchPlanner(executor, config, moduleManager, functionCatalog, catalogManager)
   }
 
+  override def loadPlan(planReference: PlanReference): CompiledPlanInternal = {
+    throw new UnsupportedOperationException(
+      "The batch planner doesn't support the persisted plan feature.")
```

Comment:
nit: `batch planner` is an internal term, how about `Compiled plans are not supported in batch mode.`

Reply:
also for the exception below